File: travel-service/src/main/java/it/roadies/travel_service/data/entity/FavouriteList.java
package it.roadies.travel_service.data.entity;

import it.roadies.travel_service.data.entity.enumerations.Visibility;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Data
@Table(name = "FAVOURITE_LIST")
public class FavouriteList {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "owner_id", nullable = false)
    private String ownerId;

    @Column(nullable = false, length = 100)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Visibility visibility = Visibility.PUBLIC;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "list", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<FavouriteListItem> items;

    @OneToMany(mappedBy = "list", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<FavouriteListShared> sharedWith;
}

File: android-app/app/src/main/java/it/roadies/android_app/model/FavouriteList.kt
package it.roadies.android_app.model
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID
@Entity(tableName = "favourite_list")
data class FavouriteList(
    @PrimaryKey val id: UUID,
    @ColumnInfo(name = "owner_id") val ownerId: String,
    @ColumnInfo(name = "name") val name: String,
    @ColumnInfo(name = "length") val length: Int,
    @ColumnInfo(name = "visibility") val visibility: String
)